contracts/dev-utils/contracts/src/OrderValidationUtils.sol

```diff
@@ -185,4 +197,77 @@ contract OrderValidationUtils is
         transferableAssetAmount = LibSafeMath.min256(balance, allowance);
         return transferableAssetAmount;
     }
+
+    /// @dev This function handles the edge cases around taker validation. This function
+    ///      currently attempts to find duplicate ERC721 token's in the taker
+    ///      multiAssetData.
+    /// @param assetData The asset data that should be validated.
+    /// @return Whether or not the order should be considered valid.
+    function _isAssetDataValid(bytes memory assetData)
+        internal
+        pure
+        returns (bool)
+    {
+        // Asset data must be composed of an asset proxy Id and a bytes segment with
+        // a length divisible by 32.
+        if (assetData.length % 32 != 4) {
+            return false;
+        }
+
+        // Only process the taker asset data if it is multiAssetData.
+        bytes4 assetProxyId = assetData.readBytes4(0);
+        if (assetProxyId != IAssetData(address(0)).MultiAsset.selector) {
+            return true;
+        }
+
+        // Get array of values and array of assetDatas
+        (, uint256[] memory assetAmounts, bytes[] memory nestedAssetData) = decodeMultiAssetData(assetData);
+
+        uint256 length = nestedAssetData.length;
+        for (uint256 i = 0; i != length; i++) {
+            // NOTE(jalextowle): As an optimization, we will break out of this function
+            // as soon as it is determined that there are no possible ERC721 or ERC1155 duplicates.
+            bool hasSeenERC721 = false;
+            bool hasSeenERC1155 = false;
+
+            bytes4 nestedAssetProxyId = nestedAssetData[i].readBytes4(0);
+            if (nestedAssetProxyId == IAssetData(address(0)).ERC721Token.selector) {
+                hasSeenERC721 = true;
+                if (_isAssetDataDuplicated(nestedAssetData, i)) {
+                    return false;
+                }
+            } else if (nestedAssetProxyId == IAssetData(address(0)).ERC1155Assets.selector) {
+                hasSeenERC1155 = true;
+                if (_isAssetDataDuplicated(nestedAssetData, i)) {
+                    return false;
+                }
+            }
+
+            if (!hasSeenERC721 && !hasSeenERC1155) {
+                break;
+            }
```

Comment:
What about the case where the taker asset is MAP(ERC20, Duplicate721, Duplicate721) ?

Reply:
Good point. I'm just going to remove this for now.